gym_electric_motor/callbacks.py

```diff
@@ -0,0 +1,98 @@
+            step_size(float): The value by which each limit gets updated at each step
+            update_time(string): When the update happens. "step" for the end of a step, "episode" for the end of an episode
+            update_freq(int): After how many cumulative units of update_time an update occurs
+        
+        Additional Notes:
+            All limit_margins should be between -1 and 1
+
+        """
+        assert update_time in ['step', 'episode'], "Chose an option of either 'step' or 'episode' for updating on cumultative steps or episodes"
+        assert initial_limit_margin[1] > initial_limit_margin[0], "First element of limit margin has to be smaller than second"
+        assert maximum_limit_margin[1] > maximum_limit_margin[0], "First element of limit margin has to be smaller than second"
+        assert initial_limit_margin[0] >= -1, "Lower limit margin has to be bigger than or equal to -1"
+        assert maximum_limit_margin[0] >= -1, "Lower limit margin has to be bigger than or equal to -1"
+        assert initial_limit_margin[1] <= 1, "Upper limit margin has to be smaller than or equal to 1"
+        assert maximum_limit_margin[1] <= 1, "Upper limit margin has to be smaller than or equal to 1"
+        
+        self._limit_margin = initial_limit_margin
+        self._maximum_limit_margin = maximum_limit_margin
+        self._step_size = step_size
+        self._update_time = 0 if update_time == 'step' else 1
+        if self._update_time == 0:
+            self._step = 0
+        else:
+            self._episode = 0
+        self._update_freq = update_freq
+        
+    def set_env(self, env):
+        #see docstring of superclass
+        #assertions added to check for the right reference generator 
+        if env._reference_generator.__class__ == SwitchedReferenceGenerator:
+            for sub_generator in env._reference_generator._sub_generators:
+                assert issubclass(sub_generator.__class__, SubepisodedReferenceGenerator), ("The AdaptiveLimitMargin does only support the SubepisodedReferenceGenerator as reference generator or "
+                                                                                            "SwitchedReferenceGenerator with SubepisodedReferenceGenerator as all sub reference generators")
+        else:
+            assert issubclass(env._reference_generator.__class__, SubepisodedReferenceGenerator), ("The AdaptiveLimitMargin does only support the SubepisodedReferenceGenerator as reference generator or" 
+                                                                                                 "SwitchedReferenceGenerator with SubepisodedReferenceGenerator as all sub reference generators")
+        self._env = env
+        #initial image margin added to the reference generator
+        if env._reference_generator.__class__ == SwitchedReferenceGenerator:
```

Comment:
Is using type() safe too? Since for the following line I need to determine the class of my object to check for its inheritance:

```
assert issubclass(type(sub_generator), SubepisodedReferenceGenerator)
```